database/migrations/2021_01_20_031335_add_room_type_to_rooms_table.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddRoomTypeToRoomsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('rooms', function (Blueprint $table) {
+            $table->string('room_type')->default('');
+        });
```

Comment:
if you have a const list of room types, why is the default empty string?

Reply:
i got an error when i tried to make it null so i changed it default and it worked fine.

Reply:
if you want the possibility of this being null then add `->nullable()`.

Reply:
right now the db will refuse to take null values unless nullable is specified

Reply:
actually, the goal is that this is a NOT NULL field with no defaults. which is what failed when removing `->default('')`
postgres did not appreciate this extra validation